### ISSUE TITLE
Fix parse error for sourcemaps in JavaScript

### ISSUE THREAD

**Issue by brandonkal** [stylelint/stylelint#4282]
> Clearly describe the bug

Stylelint misinterprets Javascript code that builds a source map comment as the source map for the Javascript file itself.

> Which rule, if any, is the bug related to?

Parse error, so all rules.

> What CSS is needed to reproduce the bug?

```js
let map = 'demo'
let cssText = ''
cssText += `/*# sourceMappingURL=data:application/json;base64,${map}*/`
```

> What stylelint configuration is needed to reproduce the bug?

Any

> Which version of stylelint are you using?

`9.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`stylelint broken.ts`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

CSS-inJS

> What did you expect to happen?

No attempt to read the source map.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
SyntaxError: Unexpected token � in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.parseSourceMapInput (<<DIRNAME>>/node_modules/source-map/lib/util.js:433:15)
    at new SourceMapConsumer (<<DIRNAME>>/node_modules/source-map/lib/source-map-consumer.js:17:22)
    at PreviousMap.consumer (<<DIRNAME>>/node_modules/postcss/lib/previous-map.js:69:28)
    at new Input (<<DIRNAME>>/node_modules/postcss/lib/input.js:93:22)
    at parseStyle (<<DIRNAME>>/node_modules/postcss-syntax/parse-style.js:118:10)
    at processor (<<DIRNAME>>/node_modules/postcss-syntax/processor.js:26:9)
    at Object.parse (<<DIRNAME>>/node_modules/postcss-syntax/parse.js:25:10)
    at new LazyResult (<<DIRNAME>>/node_modules/postcss/lib/lazy-result.js:60:16)
    at getCode.then.code (<<DIRNAME>>/node_modules/stylelint/lib/getPostcssResult.js:128:22)
```

**Comment by evilebottnawi** [comment]
Looks like it is bug in postcss

**Comment by alexandrzavalii** [comment]
Still have this error with latest `"stylelint": "13.2.0" `
Is there a workaround for this?

**Comment by jeddy3** [comment]
I've added an "upstream" label as the issue appears to be in PostCSS.

There was a pull request in progress (https://github.com/postcss/postcss/pull/1294), but it seems it was closed because of a miscommunication. @alexandrzavalii If you have time, feel free to duplicate that pull request and adapt the regex so that the integration tests pass. That will fix the issue here in stylelint.

**Comment by brandonkal** [comment]
I don't have the cycles to work on that now @alexander-schranz so go ahead and adapt my PR.

**Comment by alexander-schranz** [comment]
@brandonkal think you wanted to mark @alexandrzavalii

**Comment by brandonkal** [comment]
Yes. Sorry about that!

**Comment by emzoumpo** [comment]
Hi @brandonkal, I was also experiencing the same issue and inspired by your fix made a new PR on PostCSS (https://github.com/postcss/postcss/pull/1360)

**Comment by emzoumpo** [comment]
Hi @brandonkal, my pr was merged and released in PostCSS 7.0.31, could you verify if the issue is resolved now? Thanks!
